liver_imaging_analysis/models/liver_segmentation.py

```diff
@@ -680,75 +682,85 @@ def train_liver(
-        if pretrained is true. If not defined, the potential 
+        determines the path of the checkpoint to be loaded
+        if pretrained is true. If not defined, the potential
         cp path will be loaded from config.
     epochs : int
         number of training epochs.
         If not defined, epochs will be loaded from config.
     evaluate_epochs : int
         The number of epochs to evaluate model after. Default is 1.
     batch_callback_epochs : int
-        The frequency at which per_batch_callback will be called. 
+        The frequency at which per_batch_callback will be called.
         Expects a number of epochs. Default is 100.
     save_weight : bool
         whether to save weights or not. Default is True.
     save_path : str
         the path to save weights at if save_weights is True.
-        If not defined, the potential cp path will be loaded 
+        If not defined, the potential cp path will be loaded
         from config.
     test_batch_callback : bool
         whether to call per_batch_callback during testing or not.
         Default is False
     """
-    if cp_path is None:
-        cp_path = config.save["potential_checkpoint"]
-    if epochs is None:
-        epochs = config.training["epochs"]
-    if save_path is None:
-        save_path = config.save["potential_checkpoint"]
+
     set_seed()
     model = LiverSegmentation(modality, inference)
     model.load_data()
+
+    # checkpoints save path
+    cp_dir = os.path.join(
+        config.save["potential_checkpoint"],
+        config.name["experiment_name"],
+        config.name["run_name"],
+        "",
+    )
+    # Check if the directory exists and create it if not
+    if not os.path.exists(cp_dir):
+        os.makedirs(cp_dir)
+    save_path = f"{cp_dir}/potential_checkpoint"
+
     model.data_status()
-    if pretrained:
-        model.load_checkpoint(cp_path)
     model.compile_status()
-    init_loss, init_metric = model.test(
-                                model.test_dataloader, 
-                                callback = test_batch_callback
-                                )
-    print(
-        "Initial test loss:", 
-        init_loss,
-        )
-    print(
-        "\nInitial test metric:", 
-        init_metric.mean().item(),
-        )
-    model.fit(
-        epochs = epochs,
-        evaluate_epochs = evaluate_epochs,
-        batch_callback_epochs = batch_callback_epochs,
-        save_weight = save_weight,
-        save_path = save_path
-    )
-    # Evaluate on latest saved check point
-    model.load_checkpoint(save_path)
-    final_loss, final_metric = model.test(
-                                model.test_dataloader, 
-                                callback = test_batch_callback
-                                )
-    print(
-        "Final test loss:", 
-        final_loss,
+
+    if automate == True:
+        # Create an Optuna study
+        study = optuna.create_study(direction="minimize")
```

Comment:
remember not to make this direction param hard-coded

Reply:
done